airflow/www/views.py

```diff
@@ -771,6 +773,29 @@ def _get_log(self, ti, log_filename):
 
         return log
 
+    @expose('/get_log_js')
+    @login_required
+    @wwwutils.action_logging
+    def get_log_js(self):
+        """ JavaScript endpoint for fetching logs. """
+        logging_backend = cached_logging_backend()
+        if not logging_backend:
+            raise AirflowException("Logging backend not found.")
+
+        dag_id = request.args.get('dag_id')
+        task_id = request.args.get('task_id')
+        execution_date = request.args.get('execution_date')
+        dttm = dateutil.parser.parse(execution_date)
```

Comment:
Indeed, it would be really nice if we have some sort of unit test written for front-end and Flask side.